### PR TITLE
Add EventListener::OnExternalFileIngested() event

### PR DESCRIPTION
**Pull request by IslamAbdelRahman**
Add EventListener::OnExternalFileIngested() to allow user to subscribe to external file ingestion events

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
Just curious, what's purpose of the new event?

File: db/db_impl.cc [hunk 1]
```diff
@@ -6539,9 +6539,35 @@ Status DBImpl::IngestExternalFile(
   // Cleanup
   ingestion_job.Cleanup(status);
 
+  if (status.ok()) {
```
Comment (yiwu-arbug) [inline db/db_impl.cc]: Here we trigger the event for all files at the end. Do we want to notify listener immediately after each file is ingested?

Comment (IslamAbdelRahman) [inline db/db_impl.cc]: Actually all files are ingested in one atomic batch (everything is copied first) and then atomically we ingest all of them at once.

**Comment by IslamAbdelRahman** [comment]
@yiwu-arbug, MyRocks build it's optimizer statistics based on the SST files statistics in RocksDB. they already subscribe to Flush events to update there statistics when a new file is created. but now they cannot get this statistics when they ingest data using IngestExternalFile()

**Comment by IslamAbdelRahman** [comment]
rebase

**Pull request closed without merging**